src/GestureManager.cpp

```diff
@@ -243,8 +254,24 @@ void GestureManager::dragGestureUpdate(const wf::touch::gesture_event_t& ev) {
     }
 
     switch (this->getActiveDragGesture()->type) {
-        case DragGestureType::SWIPE:
-            this->updateWorkspaceSwipe();
+        case DragGestureType::SWIPE: {
+            if (this->hookHandled) {
+                auto gev = this->getActiveDragGesture().value();
+
+                const auto monArea = this->getMonitorArea();
+                const auto delta_percent =
+                    this->m_sGestureState.get_center().delta() / wf::touch::point_t(monArea.w, monArea.h);
+
+                IPointer::SSwipeUpdateEvent ev = {
+                    .fingers = (uint32_t)gev.finger_count,
+                    .delta   = Vector2D(delta_percent.x * 5, delta_percent.y * 5),
```

Comment:
To make it easier to hook onto, I convert the touch gesture event to a touchpad gesture event. However, I don't really know how to properly convert the delta.

Do you have any ideas @horriblename

Reply:
I'm not really sure what unit touchpad swipes are in (and I have no clue where to even dig up this info). you can either:
1. try whatever wizardry I did in updateWorkspaceSwipe()
2. ask vaxry
3. convert to touch events instead, in this case you can use `GestureManager::pixelPositionToPercentagePosition()` to get the position

try the 1 and 2 first

Reply:
you can use GestureManager::pixelPositionToPercentagePosition() to get the position

Reply:
hmm, I was doing it this way to make it as easy as possible to implement in hyprexp. Is `ITouch::SMotionEvent` a Hyprland thing?

Reply:
@horriblename `ITouch::SMotionEvent` doesn't have a `fingers` prop